README.md

```diff
@@ -101,6 +101,26 @@ logger.error<Fields>(
 
 Bearer tokens are redacted regardless of their placement in the log object.
 
+#### Pino Redaction
+
+To redact other properties, use the `redact` logger options as per [pino redaction] docs.
+
+Pino has a limitation where you can either redact or remove specified `redact.paths` by setting `redact.remove` to `true` or `false`.
+
+A logger option has been added to allow you to redact some paths and remove other paths.
+
+Example:
+
+```typescript
+const logger = createLogger({
+  name: 'my-app',
+  redact: {
+    paths: ['req.headers["x-redact-this"]'],
+    removePaths: ['req.headers["x-remove-this"]'],
```

Comment:
There are some `x-envoy-*` headers likely related to internal routing but I'm not if these should be blanket added for all services.

Also, being OSS, should we add default headers for removal or is it ok since it's a SEEK opinionated logger?

Reply:
It's SEEK opinionated so I'd be fine with any configuration that is not sensitive, and I reckon headers for an OSS project like Envoy would be fine.

Do we have an example of a SEEK application that would be impaired by having `x-envoy-*` removed _by default_? We'd presumably allow them to override/disable the defaults if needed.

Reply:
I've asked cloud platform about the `x-envoy-*` headers so we'll have to wait for an answer.

Reply:
Ah apologies if I was vague here; I meant adding logic here to remove the headers from logs by default, not changing our infra to stop attaching the headers.

Reply:
@72636c : I've just pushed a change that allows us to add default paths for removal.

Caveat: Since the `defaultRemovePaths` is empty, it isn't really tested, but you can pull this branch, add a path e.g. `req.headers["x-envoy-peer-metadata"]` and see the `should remove default paths when ignoreDefaultRemovePaths is missing` test result.
_You might need to `console.log({input, log})` in the `testLog` function to observe._

You can also `ignoreDefaultRemovePaths` if you don't want the default remove paths to be added.

I had to refactor the `addDefaultRedactPathStrings` (and rename to `appendDefaultRedactAndRemove`) to get this working.

Something I'm wondering is should we only override the `censor` when it is `undefined` with the caveat that the `remove` may not work proper?

Reply:
@72636c : I've just pushed a change that allows us to add default paths for removal.

Caveat: Since the `defaultRemovePaths` is empty, it isn't really tested, but you can pull this branch, add a path e.g. `req.headers["x-envoy-peer-metadata"]` and see the `should remove default paths when ignoreDefaultRemovePaths is missing` test result.
_You might need to `console.log({input, log})` in the `testLog` function to observe._

You can also `ignoreDefaultRemovePaths` if you don't want the default remove paths to be added.

I had to refactor the `addDefaultRedactPathStrings` (and rename to `appendDefaultRedactAndRemove`) to get this working.

Something I'm wondering is should we only override the `censor` when it is `undefined` with the caveat that the `remove` may not work proper?

Reply:
> Something I'm wondering is should we only override the censor when it is undefined with the caveat that the remove may not work proper?

I've updated this so that the input `censor` is used for redaction but `removePaths` properties are still removed.